### ISSUE TITLE
Rendering the same scene to two different renderers.

### ISSUE THREAD

**Issue by georgealways** [mrdoob/three.js#189]
Hey -- 

Trying to render the same scene to two different (WebGL) renderers with two different cameras. The effect is such that only the first renderer to be asked to render the scene within my draw loop is actually drawn.

```javascript
function draw() {
debugRenderer.render(scene, debugCamera);
renderer.render(scene, camera);
}
```

Have you encountered this before?

**Comment by ekatzenstein** [comment]
This comment may be more stack overflow, but is a possible work around to clone the scene for the time being?  https://github.com/mrdoob/three.js/issues/3032

**Comment by makc** [comment]
what is the status of this + #3032 ? is it enough to scene.clone() to be able to pass it to another renderer?

**Comment by makc** [comment]
So I've tried and got similar error to #3473
![screen shot 2015-06-04 at 11 35 44](https://cloud.githubusercontent.com/assets/242577/7981297/40a288c2-0aae-11e5-97ef-c6beba0b3985.png)
At this line:
```javascript
for ( var i = 0,l = geometryGroupsList.length; i < l; i ++ ) {
```
So I guess the problem is that scene.clone() is not deep enough.

**Comment by makc** [comment]
Trying this approach:
```javascript
loader = new THREE.ObjectLoader();
scene2 = loader.parse(scene.toJSON());
```
This strips the scene of all the textures :sake:

**Comment by makc** [comment]
Okay, so in case anyone runs into this, too - this worked for me:
```javascript
		var loader = new THREE.ObjectLoader();
		var scene2 = loader.parse( scene.toJSON() );

		// we have deep-cloned scene now except for textures
		var mat2tex = {}, textures = {};
		scene.traverse(function (object) {
			if (object.material && object.material.map) {
				var textureId = object.material.map.id;
				if (!textures[textureId]) {
					// clone texture
					var texture = object.material.map;
					var texture2 = new THREE.Texture (
						texture.image,
						texture.mapping,
						texture.wrapS,
						texture.wrapT,
						texture.magFilter,
						texture.minFilter,
						texture.format,
						texture.type,
						texture.anisotropy
					);
					texture2.needsUpdate = true;
					textures[textureId] = texture2;
				}
				mat2tex[object.material.uuid] = textureId;
			}
		});
		scene2.traverse(function (object) {
			if (object.material && (mat2tex[object.material.uuid] != undefined)) {
				object.material.map = textures[mat2tex[object.material.uuid]];
			}
		});
```

**Comment by mlepage** [comment]
I'm just learning three.js now (already know OpenGL though) and experimenting with multiple passes. I think I may eventually also want to render one scene into multiple textures (still figuring things out). So good to see it's still being looked that, thanks Mr. Doob!

**Comment by mrdoob** [comment]
Fixed by #6723.

**Comment by prominentdetail** [comment]
@mrdoob I think I'm running into this issue:
https://discourse.threejs.org/t/render-same-scene-to-a-second-webglrenderer-causes-textures-to-disappear-and-gl-invalid-enum/23970/5
Has this issue returned- any info is appreciated.
When I render my scene to a second webrenderer the textures disappear, but the wireframes and lines and points get rendered.. why would that be?
